### PR TITLE
Cities dev

### PR DESCRIPTION
**Pull request by jjgomezcadenas**
This PR simplifies the structure of the cities, making then one step closer to the goal of being truly invisibles. The most important features are:

1) Most cities have the same run() method, that now goes to the base (only Zaira needs to re-write the run() method).

2) Cities reading RWF (Isidora and Irene) have the same file_loop() method, that now goes to their common base.

3) Cities reading PMAPS (Dorothea and Penthiselea) have the same file_loop() method that now goes to their common base (Dorothea and Penthiselea both inherit now from HitCity).

4) A counter class allows easy bookeping and common go() method. 

Overall, the concrete cities are much thiner and simpler, with considerable reduction of noise and sizeable repayment of TD).

### CODE REVIEW

File: invisible_cities/core/ic_types_test.py [hunk 1]
```diff
@@ -140,3 +141,82 @@ def test_cxy(xyc, a, b):
     np.isclose (xyc.R  ,   r, rtol=1e-4)
     np.isclose (xyc.Phi, phi, rtol=1e-4)
     np.allclose(xyc.pos, pos, rtol=1e-3, atol=1e-03)
+
+def test_counters():
```
Comment (jacg) [inline invisible_cities/core/ic_types_test.py:145]: Please break this test up into separate orthogonal tests, wherever possible.

File: invisible_cities/cities/dorothea.py [hunk 2]
```diff
@@ -44,61 +43,75 @@ def __init__(self, **kwds):
...
-            event_numbers, timestamps = self.event_numbers_and_timestamps_from_file_name(filename)
-
-            max_events_reached = self.event_loop(event_numbers, timestamps, write_kr, s1_dict, s2_dict, s2si_dict)
-
-            if max_events_reached:
-                print('Max events reached')
-                break
-            else:
-                print("OK")
+    # def file_loop(self):
```
Comment (jacg) [inline invisible_cities/cities/dorothea.py]: Don't forget to remove these commented-out lines before the final merge.

File: invisible_cities/cities/penthesilea.py [hunk 3]
```diff
@@ -14,87 +22,131 @@
...
-                break
-            else:
-                print("OK")
-
-        return nevt_in, nevt_out
-
-    def _event_loop(self, event_numbers, timestamps,
-                    nevt_in, nevt_out, write_hits, S1_dict, S2_dict, S2Si_dict):
-        max_events_reached = False
+    # def run(self):
```
Comment (jacg) [inline invisible_cities/cities/penthesilea.py]: More commented-out code here to be removed before final commit.

**Comment by jacg** [comment]
I'm sorry I don't have time to look at this in detail, but I like the general direction in which it is heading.

**Review by jmbenlloch — APPROVED** [review]
Important refactoring of the cities and general improvements. @abotas has been using this code in some notebooks and everything seems to work well.

**Pull request merged**